src/coreclr/jit/target.h

```diff
@@ -208,8 +208,163 @@ enum _regMask_enum : unsigned
 // In any case, we believe that is OK to freely cast between these types; no information will
 // be lost.
 
-#if defined(TARGET_AMD64) || defined(TARGET_ARMARCH) || defined(TARGET_LOONGARCH64) || defined(TARGET_RISCV64)
+typedef _regNumber_enum regNumber;
+typedef unsigned char   regNumberSmall;
+
+#if defined(TARGET_AMD64) || defined(TARGET_ARM) || defined(TARGET_LOONGARCH64) || defined(TARGET_RISCV64)
 typedef unsigned __int64 regMaskTP;
+#elif defined(TARGET_ARM64)
+typedef unsigned __int64 regMaskSmall;
+struct _regMaskAll
+{
+    unsigned __int64 low;
+
+    _regMaskAll(unsigned __int64 regMask)
+        : low(regMask)
+    {
+    }
+
+    _regMaskAll()
+        : low(RBM_NONE)
+    {
+    }
+
+    FORCEINLINE operator _regMaskAll() const
+    {
+        return _regMaskAll{static_cast<uint64_t>(low)};
+    }
+
+    FORCEINLINE explicit operator bool() const
+    {
+        return low != 0;
+    }
+
+    FORCEINLINE explicit operator regMaskSmall() const
+    {
+        return (regMaskSmall)low;
+    }
+
+    FORCEINLINE explicit operator unsigned int() const
+    {
+        return (unsigned int)low;
+    }
+
+    FORCEINLINE unsigned __int64 getLow() const
+    {
+        return low;
+    }
+};
```

Comment:
I would remove all the `FORCEINLINE` and leave that up to the compiler, unless we actually see it as being necessary.

Reply:
Sure, tried removing and it didn't make any difference.

Reply:
I don't see them removed. If it isn't necessary then I don't think we should be marking them.